src/CacheMiddleware.php

```diff
@@ -115,10 +124,12 @@ public function __invoke(callable $handler)
             if (!isset($this->httpMethods[strtoupper($request->getMethod())])) {
                 // No caching for this method allowed
 
-                return $handler($request, $options)->then(
+                return $handler($request, $options)->then(  
                     function (ResponseInterface $response) use ($request) {
-                        // Invalidate cache after a call of non-safe method on the same URI
-                        $response = $this->invalidateCache($request, $response);
+                        if (!isset($this->safeMethods[$request->getMethod()])) {
```

Comment:
why not `protected $safeMethods = ['GET', 'HEAD', 'OPTIONS', 'TRACE'];` and `if (!in_array($request->getMethod(), $this->safeMethods)) {`?

Reply:
@xavierleune was pointed out in_array was slower and inconsistent with the structure of the httpMethods array

Reply:
@xavierleune @rhysemmerson oh sorry I didn't realise `safeMethods` or `httpMethods` had thousands of items in it and it's executed an exponential amount of times every request. 

If I was responsible for this project I would give readability more importance than disobeying fundamental programming principles for the sake of saving 5 cpu ticks. But i'm not responsible so thanks for reading and have a nice day 🙌🏻

Reply:
@hannesvdvreken not sure to understand why you're using this tone (nor to like it). I made a suggestion, I'm just a (new) enthusiast user. Sorry to try to involve in an open source project I find valuable.
But to make my case: consistency is important to me.

Whatever is fine for the team works for me.